compiler/error/src/lib.rs

```diff
@@ -0,0 +1,554 @@
+            code
+        }
+    }
+
+    pub fn builder() -> SyntaxErrorBuilder<NotSet, NotSet, NotSet, NotSet, NotSet> {
+        SyntaxErrorBuilder::new()
+    }
+
+    /** Prints the error to stderr with the default context
+          # Panic
+          May panic if the area is referencing to locations that don't exist in the provided code
+    */
+    pub fn print_to_stderr_default_context(&self) {
+        self.write(ErrorOutputConfig::default())
+    }
+
+    /** Writes the error to the specified output
+       # Panic
+       May panic if the area is referencing to locations that don't exist in the provided code
+    */
+    pub fn write(&self, print_config: ErrorOutputConfig) {
+        let error_context = print_config.context();
+        // Orange
+        let error_code_color = Color::BrightWhite;
+        let error_msg_color = Color::Red;
+        let syntax_error_color = Color::BrightWhite;
+        let code_color = Color::Primary;
+
+        let mut line_starting_pos = 0;
+        // The start and end
+        let error_start_line = self.area.start().line();
+        let error_end_line = self.area.end().line();
+        // The start and end including the padding
+        let display_end = error_end_line + error_context;
+        let display_start = if error_start_line < error_context {
+            0
+        } else {
+            error_start_line - error_context
+        };
+
+        // The processed lines with annotations
+        let mut lines = Vec::new();
+        // Where the error begins, relative to the start of code
+        let mut error_start_char = 0;
+        let mut error_end_char = 0;
+        // Process the codelines
+        for (line_num, line) in self.code.lines().enumerate() {
+            // If the code line is out of the window to print, skip it
+            if line_num < display_start || line_num > display_end {
+                //
+                Self::update_line_starting_pos(&self.code, &mut line_starting_pos);
+                continue;
+            }
+            // Update the error start and end
+            if line_num == error_start_line {
+                error_start_char = line_starting_pos + self.area.start().char();
+            }
+            if line_num == error_end_line {
+                error_end_char = line_starting_pos + line.len() + self.area.end().char();
+            }
+            // Where the error begins and where it ends in the current line
+            // Used for making the text white or yellow
+            #[allow(clippy::comparison_chain)]
+            // Lines without annotations always have the lf selected
+            let line_error_start_char = if line_num < error_start_line || line_num > error_end_line
+            {
+                line.len()
+            }
+            //Before the error lines, so it doesn't begin at all
+            else if line_num == error_start_line {
+                self.area.start().char()
+            } else {
+                0
+            };
```

Comment:
I don't really understand why we avoid a match statement there. I find the match expression much cleaner.


```suggestion
let line_error_start_char = match line_num {
    n if n < error_start_line || n > error_end_line => line.len(),
    n if n == error_start_line => self.area.start().char(),
    _ => 0,
};

```